site/pages/api/create-consequence-journeys.api.ts

```diff
@@ -0,0 +1,218 @@
+import { Journey, JourneysConsequence, Service } from "@create-disruptions-data/shared-ts/disruptionTypes";
+import { journeysConsequenceSchema } from "@create-disruptions-data/shared-ts/disruptionTypes.zod";
+import { Datasource } from "@create-disruptions-data/shared-ts/enums";
+import { NextApiRequest, NextApiResponse } from "next";
+import {
+    COOKIES_CONSEQUENCE_JOURNEYS_ERRORS,
+    CREATE_CONSEQUENCE_JOURNEYS_PATH,
+    DASHBOARD_PAGE_PATH,
+    DISRUPTION_DETAIL_PAGE_PATH,
+    REVIEW_DISRUPTION_PAGE_PATH,
+    TYPE_OF_CONSEQUENCE_PAGE_PATH,
+} from "../../constants";
+import { getNocCodesForOperatorOrg } from "../../data/dynamo";
+import { TooManyConsequencesError } from "../../errors";
+import { flattenZodErrors, getLargestConsequenceIndex } from "../../utils";
+import {
+    destroyCookieOnResponseObject,
+    getReturnPage,
+    handleUpsertConsequence,
+    isDisruptionFromTemplate,
+    redirectTo,
+    redirectToError,
+    redirectToWithQueryParams,
+    setCookieOnResponseObject,
+} from "../../utils/apiUtils";
+import { getSession } from "../../utils/apiUtils/auth";
+
+export const formatCreateConsequenceJourneysServicesBody = (body: object) => {
+    const services = Object.entries(body)
+        .filter((item) => item.toString().startsWith("service"))
+        .map((arr: string[]) => {
+            const [, values] = arr;
+            return JSON.parse(values) as Service;
+        });
+
+    const cleansedBody = Object.fromEntries(
+        Object.entries(body).filter(
+            (item) => !item.toString().startsWith("journey") && !item.toString().startsWith("service"),
+        ),
+    );
+
+    const journeys = Object.entries(body)
+        .filter((item) => item.toString().startsWith("journey"))
+        .map((arr: string[]) => {
+            const [, values] = arr;
+            return JSON.parse(values) as Journey;
+        });
+
+    return {
+        ...cleansedBody,
+        services,
+        journeys,
+    };
+};
+
+const createConsequenceJourneys = async (req: NextApiRequest, res: NextApiResponse): Promise<void> => {
+    try {
+        const queryParam = getReturnPage(req);
+        const isFromTemplate = isDisruptionFromTemplate(req);
+
+        const { template, addAnotherConsequence } = req.query;
+
+        const body = req.body as JourneysConsequence;
+
+        const formattedBody = formatCreateConsequenceJourneysServicesBody(body);
+
+        const validatedBody = journeysConsequenceSchema.safeParse(formattedBody);
+
+        const session = getSession(req);
+
+        const { draft } = req.query;
+
+        if (!session) {
+            throw new Error("No session found");
+        }
+
+        if (!validatedBody.success) {
+            if (!body.disruptionId || !body.consequenceIndex) {
+                throw new Error("No disruptionId or consequenceIndex found");
+            }
+
+            setCookieOnResponseObject(
```

Comment:
When I force an error (e.g. leave the consequence description blank) and try to submit the consequence, when the page loads with the error message the journeys I selected gets cleared.

Reply:
done